dropshot/src/server.rs

```diff
@@ -147,11 +157,13 @@ impl<C: ServerContext> HttpServerStarter<C> {
                         api,
                         private,
                         log,
+                        handler_waitgroup.worker(),
```

Comment:
This is probably not something to worry about right now, but: I wonder how debuggable this is, either _in situ_ or _post mortem_.  Like if you walk up to a server that's shutting down and stuck waiting on one of these, would you have any way to know which request it was waiting on?  I imagine eventually we'll want to elevate this to an API but that's probably way down the road.

Just to show what I mean, in the past I built something like this:
https://github.com/TritonDataCenter/node-vasync#barrier-coordinate-multiple-concurrent-operations
In that thing, each outstanding operation has a distinct name.  You can take the whole object and expose that over a debug HTTP API to ask the server "what are the operations you're waiting on".  This proved incredibly useful.  But that was more for stuff like our own RSS, where you've got complicated long-running things that could get stuck somewhere.  This particular case seems less likely to be a problem.  It would just be neat to have a thing like waitgroup but where it was easy to ask it what it was waiting on.

Reply:
> This is probably not something to worry about right now, but: I wonder how debuggable this is, either in situ or post mortem. Like if you walk up to a server that's shutting down and stuck waiting on one of these, would you have any way to know which request it was waiting on?

Probably not! Internally the waitgroup is just a wrapper around an [AtomicWaker](https://docs.rs/atomic-waker/1.1.1/atomic_waker/struct.AtomicWaker.html), which itself is just a glorified `AtomicUsize`; I think you could pretty quickly find the _count_, but assuming it's something like `1`, I don't know how you'd find the guilty party.